src/expr/src/expr/agg.rs

```diff
@@ -33,6 +33,7 @@ pub enum AggKind {
     ApproxCountDistinct,
     ArrayAgg,
     FirstValue,
+    SinglePhaseAppendOnlyApproxDistinct,
```

Comment:
Is this new agg kind really necessary? I mean, other agg kinds that support both 1-phase or 2-phase and append-only or non-append-only still only need one agg kind variant to represent.

Reply:
> Is this new agg kind really necessary? I mean, other agg kinds that support both 1-phase or 2-phase and append-only or non-append-only still only need one agg kind variant to represent.

States of `ApproxCountDistinct` and `SinglePhaseAppendOnlyApproxDistinct` are different.

Reply:
> Is this new agg kind really necessary? I mean, other agg kinds that support both 1-phase or 2-phase and append-only or non-append-only still only need one agg kind variant to represent.

max(max()) is max, sum(sum()) is sum, count(count()) is not count but sum(count()) is count, however, there doesn't exist an X which satisfies that X(approx_count_distinct()) is approx_count_distinct

Reply:
> Is this new agg kind really necessary? I mean, other agg kinds that support both 1-phase or 2-phase and append-only or non-append-only still only need one agg kind variant to represent.

From another perspective, result of other aggs is unique, but result of the new `SinglePhaseAppendOnlyApproxDistinct` agg may differ from the original `ApproxCountDistinct` agg. (Because we previously use `initial_count`)

Reply:
> Is this new agg kind really necessary? I mean, other agg kinds that support both 1-phase or 2-phase and append-only or non-append-only still only need one agg kind variant to represent.

fixed. just removed `SinglePhaseAppendOnlyApproxDistinct`